x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/AsyncSearchTask.java

```diff
@@ -338,7 +341,7 @@ protected void onQueryFailure(int shardIndex, SearchShardTarget shardTarget, Exc
             // best effort to cancel expired tasks
             checkCancellation();
             searchResponse.get().addShardFailure(shardIndex,
-                new ShardSearchFailure(exc, shardTarget.getNodeId() != null ? shardTarget : null));
+                new ShardSearchFailure(exc, shardTarget.getNodeIdText() != null ? shardTarget : null));
```

Comment:
this one was causing NPE? lovely to get NPE as part of a null check :) maybe getNodeId should be changed to return null instead of NPE when nodeId is null? it seems like a trappy behaviour.

Reply:
+1, I pushed https://github.com/elastic/elasticsearch/pull/55435/commits/47f35443ad45f0fb2ac4959761c99ff195be22ac to adapt the code in the shard search target class directly.